low-level/files/opt/seagate/sspl/setup/cortx_manifest/lr2/manifest_storage.py

```diff
@@ -0,0 +1,381 @@
+                "name": expander.get("name", "NA"),
+                "location": expander.get("location", "NA"),
+                "drawer_id": expander.get("drawer-id", "NA"),
+                "status": expander.get("status", "NA"),
+                "fw-revision": expander.get("fw-revision", "NA"),
+            }]
+        }
+
+    def get_sideplane_expander_info(self):
+        """Update and return sideplane information in specific format."""
+        sideplane_expander_list = []
+        data = []
+        enclosures = self.get_realstor_encl_data("enclosures")
+        #TODO : Handle CORVAULT sideplane expander data without expecting drawers
+        encl_drawers = enclosures[0].get("drawers")
+        if encl_drawers:
+            for drawer in encl_drawers:
+                sideplanes = drawer.get("sideplanes")
+                sideplane_expander_list.extend(sideplanes)
+        for sideplane in sideplane_expander_list:
+            sideplane_dict = {
+                "uid": sideplane.get("durable-id", "NA"),
+                "type": sideplane.get("type", "NA"),
+                "description": sideplane.get("description", "NA"),
+                "product": sideplane.get("object-name", "NA"),
+                "manufacturer": sideplane.get("vendor", "NA"),
+                "serial_number": sideplane.get("serial-number", "NA"),
+                "version": sideplane.get("hardware-version", "NA"),
+                "part_number": sideplane.get("part-number", "NA"),
+                "last_updated": int(time.time()),
+                "specifics": [{
+                    "name": sideplane.get("name", "NA"),
+                    "location": sideplane.get("location", "NA"),
+                    "drawer_id": sideplane.get("drawer-id", "NA"),
+                    "status": sideplane.get("status", "NA"),
+                    "expanders": [self.get_expander_data(expander) \
+                        for expander in sideplane['expanders']]
+                }]
+            }
+            data.append(sideplane_dict)
+            logger.debug(self.log.svc_log(
+                f"Sideplane Manifest Data:{data}"))
+        return data
+
+    @staticmethod
+    def get_realstor_encl_data(fru: str):
+        """Fetch fru information through webservice API."""
+        fru_data = []
+        fru_uri_map = {
+            "enclosures": ENCL.URI_CLIAPI_SHOWENCLOSURE,
+            "controllers": ENCL.URI_CLIAPI_SHOWCONTROLLERS,
+            "power-supplies": ENCL.URI_CLIAPI_SHOWPSUS,
+            "fan-modules": ENCL.URI_CLIAPI_SHOWFANMODULES,
+            "disks": ENCL.URI_CLIAPI_SHOWDISKS,
+            "frus": ENCL.URI_CLIAPI_SHOWFRUS
+        }
+        url = ENCL.build_url(fru_uri_map.get(fru))
+        response = ENCL.ws_request(url, ENCL.ws.HTTP_GET)
+        if fru == "frus":
+            fru = "enclosure-fru"
+        elif fru == "disks":
+            fru = "drives"
+        if not response or response.status_code != ENCL.ws.HTTP_OK:
+            return []
```

Comment:
please log reponse and status code if the response is not 200.

Reply:
done